src/cml.js

```diff
@@ -330,11 +330,16 @@ class CML {
   }
 
   async ci(opts = {}) {
-    const { userEmail = GIT_USER_EMAIL, userName = GIT_USER_NAME } = opts;
+    const {
+      unshallow = false,
+      userEmail = GIT_USER_EMAIL,
+      userName = GIT_USER_NAME
+    } = opts;
 
     const driver = getDriver(this);
-    const command = await driver.updateGitConfig({ userName, userEmail });
-    await exec(command);
+    const shallow = await exec('git rev-parse --is-shallow-repository');
+    if (unshallow && shallow === 'true') await exec('git fetch --unshallow');
+    await exec(await driver.updateGitConfig({ userName, userEmail }));
```

Comment:
weird to interleave ordering?

```suggestion
    await exec(await driver.updateGitConfig({ userName, userEmail }));
    const shallow = await exec('git rev-parse --is-shallow-repository');
    if (unshallow && shallow === 'true') await exec('git fetch --unshallow');
```

Reply:
It was just to put the `await exec` together. The OCD I don't have,™ you know.

Reply:
Fron what I have tested, order doesn't matter.